src/views/RecIM/components/QuickSearch/index.jsx

```diff
@@ -0,0 +1,294 @@
+              ? getHighlightedText(suggestion.UserName.split('.')[0], highlightQuery)
+              : suggestion.UserName.split('.')[0]}
+            {suggestion.UserName.includes('.') && '.'}
+            {suggestion.UserName.includes('.') &&
+              (!suggestion.LastName.match(new RegExp(`(${highlightParse(highlightQuery)})`, 'i'))
+                ? getHighlightedText(suggestion.UserName.split('.')[1], highlightQuery)
+                : suggestion.UserName.split('.')[1])}
+          </Typography>
+        </MenuItem>
+        <Divider className={styles.suggestion_divider} />
+      </>
+    );
+  }
+
+  return (
+    <Downshift
+      // Assign reference to Downshift to state property for usage elsewhere in the component
+      ref={(downshift) => {
+        setDownshift(downshift);
+      }}
+    >
+      {({ getInputProps, getItemProps, isOpen }) => (
+        <span className={styles.quick_search} key="suggestion-list-span">
+          {renderInput(
+            getInputProps({
+              placeholder: placeholder,
+              ...(isOnline
+                ? {
+                    onChange: (event) => updateQuery(event.target.value),
+                    onKeyDown: (event) => handleKeys(event.key),
+                    onBlur: () => setQuery(''),
+                  }
+                : {
+                    style: { color: 'white' },
+                    disabled: { isOnline },
+                  }),
+            }),
+          )}
+          {isOpen && query.length >= MIN_QUERY_LENGTH && (
+            <Paper square className={`${styles.dropdown} gc360_quick_search_dropdown`}>
+              {suggestions.length > 0 ? (
+                suggestions.map((suggestion) =>
+                  renderSuggestion({
+                    suggestion,
+                    itemProps: getItemProps({
+                      item: suggestion.UserName,
+                      ...(!disableLink
+                        ? {
+                            component: Link,
+                            to: `/profile/${suggestion.UserName}`,
+                          }
+                        : {}),
+                    }),
+                  }),
+                )
+              ) : (
+                <MenuItem className={styles.suggestion} style={{ paddingBottom: '5px' }}>
+                  <Typography className={styles.no_results} variant="body2">
+                    No results
+                  </Typography>
+                  <Typography className={styles.loading} variant="body2">
+                    Loading...
+                  </Typography>
+                </MenuItem>
+              )}
+            </Paper>
+          )}
+        </span>
+      )}
+    </Downshift>
+  );
+};
+
+GordonQuickSearch.propTypes = {
+  customPlaceholderText: PropTypes.string,
+  disableLink: PropTypes.any,
+  onSearchSubmit: PropTypes.func,
+};
+
+export default GordonQuickSearch;
```

Comment:
hmmm i don't know if we should be ripping out the entire gordon search? I can't think of another solution off the top of my head but it feels a little sketchy?

Benefits are obviously that Rec-IM stays modular but this is definitely a repeated component?
thoughts @silasw2  @cpabbot

Reply:
True. Most part is the same, except we also need to return `FirstName` and `LastName` otherwise the invite list shows `undefined undefined` for all people.
`    if (key === 'Enter' && suggestionList && suggestionList.length > 0) {
      if (sIndex === -1) {
        theChosenOne = suggestionList[0].UserName;
        firstName = suggestionList[0].firstName;
        lastName = suggestionList[0].lastName;
      } else {
        theChosenOne = suggestionList[sIndex].UserName;
        firstName = suggestionList[sIndex].firstName;
        lastName = suggestionList[sIndex].lastName;
      }
      // If prop set to disable link, then trigger the onSearchSubmit callback function
      // Else, redirect the user to the selected profile page
      disableLink
        ? onSearchSubmit(theChosenOne, firstName, lastName)`

Reply:
Right, I agree that we have a unique piece of the component, but it is, in essence, identical. I'm just wondering if we should handle it this way or if we should modify the actual GordonQuickSearch which would interlink 360 and Rec-IM further